src/parser.c

```diff
@@ -348,12 +428,15 @@ json_object *parser_data_create_object(const char *device_id, uint8_t sensor_id,
 			json_object_new_uint64(knot_value_as_uint64(value)));
 		break;
 	default:
-		goto fail;
+		has_err = true;
+		break;
 	}
 
 	json_object_array_add(json_array, data);
 	json_object_object_add(json_msg, KNOT_JSON_FIELD_DATA, json_array);
 
+	json_str = l_strdup(json_object_to_json_string(json_msg));
+	json_object_put(json_msg);
```

Comment:
Isn't it necessary to call `json_object_put` for `json_array` and `json_data`?

Reply:
No, it isn't. The ownership of those: json_array, json_data belongs to json_msg after json_object_array_add and json_object_object_add